apollo-router/src/spec/schema.rs

```diff
@@ -38,24 +42,353 @@ pub(crate) struct Schema {
+                        },
+                    )
+                })
+                .collect();
+
+            let input_field = |field: &hir::InputValueDefinition| {
+                (
+                    field.name().to_owned(),
+                    (
+                        field.ty().into(),
+                        field.default_value().and_then(parse_hir_value),
+                    ),
+                )
+            };
+            let input_types: HashMap<_, _> = compiler
+                .db
+                .input_objects()
+                .iter()
+                .map(|(name, def)| {
+                    (
+                        name.clone(),
+                        InputObjectType {
+                            name: name.clone(),
+                            fields: def
+                                .input_fields_definition()
+                                .iter()
+                                .chain(
+                                    def.extensions()
+                                        .iter()
+                                        .flat_map(|ext| ext.input_fields_definition()),
+                                )
+                                .map(input_field)
+                                .collect(),
+                        },
+                    )
+                })
+                .collect();
+            let enums = compiler
+                .db
+                .enums()
+                .iter()
+                .map(|(name, def)| {
+                    let values = def
+                        .enum_values_definition()
+                        .iter()
+                        .map(|value| value.enum_value().to_owned())
+                        .collect();
+                    (name.clone(), values)
+                })
+                .collect();
+
+            let root_operations = compiler
+                .db
+                .schema()
+                .root_operation_type_definition()
+                .iter()
+                .filter(|def| def.loc().is_some()) // exclude implict operations
+                .map(|def| {
+                    (
+                        def.operation_type().into(),
+                        if let hir::Type::Named { name, .. } = def.named_type() {
+                            name.clone()
+                        } else {
+                            // FIXME: hir::RootOperationTypeDefinition should contain
+                            // the name directly, not a `Type` enum value which happens to always
+                            // be the `Named` variant.
+                            unreachable!()
+                        },
+                    )
+                })
+                .collect();
+
+            let custom_scalars = compiler
+                .db
+                .scalars()
+                .iter()
+                .filter(|(_name, def)| !def.is_built_in())
+                .map(|(name, _def)| name.clone())
```

Comment:
`filter_map`

Reply:
It's just a matter of taste. I prefer this one 
```rust
    .filter_map(|(name, def)| {
        if !def.is_built_in() {
            Some(name.clone())
        } else {
            None
        }
    })
```

but it is very subjective so... as you wish :p